src/PooledArrays.jl

```diff
@@ -601,4 +601,7 @@ _perm(o::F, z::V) where {F, V} = Base.Order.Perm{F, V}(o, z)
 
 Base.Order.Perm(o::Base.Order.ForwardOrdering, y::PooledArray) = _perm(o, fast_sortable(y))
 
+Base.repeat(x::PooledArray, counts...) = PooledArray(RefArray(repeat(x.refs, counts...)), copy(x.invpool))
```

Comment:
copying of `pool` and `invpool` is not needed and should be avoided. Please check the implementation of other operations to see how copy-on-write is implemented. Thank you!

Reply:
Thanks for comments. Updated the function and add some tests.

Reply:
Ah - now I see. Such ambiguities are super annoying (and that is why having a detailed test coverage is crucial 😄). Thank you!